File: GCDLib/UI/About/frmAbout.vb
Namespace UI.About

    Public Class frmAbout

        Private Sub AboutForm_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load

            Panel2.Controls.Add(New ucAcknowledgements)

            Dim sProduct As String = "Standalone"
            If Reflection.Assembly.GetEntryAssembly().FullName.ToLower().Contains("arcmap") Then
                sProduct = "Addin"
            End If

            Dim ass As System.Reflection.Assembly = System.Reflection.Assembly.GetExecutingAssembly
            lblVersion.Text = String.Format("{0} {1}", sProduct, ass.GetName.Version)

            Try
                'lblRMVersion.Text = RasterManager.GetFileVersion().FileVersion.ToString()
                'lblGCDCoreVersion.Text = GCDCore.GetFileVersion().FileVersion.ToString()
            Catch ex As Exception
                naru.error.ExceptionUI.HandleException(ex)
            End Try

        End Sub

        Private Sub LinkLabel8_LinkClicked(ByVal sender As System.Object, ByVal e As System.Windows.Forms.LinkLabelLinkClickedEventArgs)
            Process.Start("http://www.essa.com/team/index.html#fp")
        End Sub

        Private Sub LinkLabel9_LinkClicked(ByVal sender As System.Object, ByVal e As System.Windows.Forms.LinkLabelLinkClickedEventArgs)
            Process.Start("http://www.essa.com/team/index.html#no")
        End Sub

        Private Sub LinkLabel6_LinkClicked(ByVal sender As System.Object, ByVal e As System.Windows.Forms.LinkLabelLinkClickedEventArgs)
            Process.Start("http://northarrowresearch.com/north-arrow-research/people/")
        End Sub

        Private Sub LinkLabel4_LinkClicked(ByVal sender As System.Object, ByVal e As System.Windows.Forms.LinkLabelLinkClickedEventArgs)
            Process.Start(My.Resources.PeopleJoeWheatonURL)
        End Sub

        'TODO ask Philip if this should be a link label, requires user to have outlook or other email service set up and is causing errors becasue of that
        Private Sub LinkLabel3_LinkClicked(ByVal sender As System.Object, ByVal e As System.Windows.Forms.LinkLabelLinkClickedEventArgs) Handles LinkLabel3.LinkClicked
            'hard coded path to GCD 6 Forum, perhaps the GCD 6 forum needs a more descriptive web address
            Process.Start("http://forum.bluezone.usu.edu/gcd/viewforum.php?f=42&sid=d221c1def27836aff25624551c6e56d1")
        End Sub

        Private Sub LinkLabel1_LinkClicked(ByVal sender As System.Object, ByVal e As System.Windows.Forms.LinkLabelLinkClickedEventArgs) Handles LinkLabel1.LinkClicked
            Process.Start(My.Resources.GCDWebSiteURL)
        End Sub

        Private Sub LinkLabel2_LinkClicked(ByVal sender As System.Object, ByVal e As System.Windows.Forms.LinkLabelLinkClickedEventArgs) Handles LinkLabel2.LinkClicked
            Process.Start(My.Resources.HelpBaseURL)
        End Sub

    End Class

End Namespace

File: GCDCore/UserInterface/About/frmAbout.cs
using System;
using System.Windows.Forms;
using GCDCore.Project;
using System.Diagnostics;

namespace GCDCore.UserInterface.About
{
    public partial class frmAbout : Form
    {
        public frmAbout()
        {
            InitializeComponent();
        }

        private void AboutForm_Load(Object sender, EventArgs e)
        {
            Text = string.Format("About {0}", Properties.Resources.ApplicationNameLong);
            webBrowser1.Url = new Uri("http://gcd.riverscapes.net/dotnetack.html");            
            lblName.Text = string.Format("{0} {1}", Properties.Resources.ApplicationNameShort, ProjectManager.IsArcMap ? "AddIn" : "Standalone");
            lblVersion.Text = System.Reflection.Assembly.GetExecutingAssembly().GetName().Version.ToString();
        }

        private void lnkJoeWheaton(System.Object sender, System.Windows.Forms.LinkLabelLinkClickedEventArgs e)
        {
            Process.Start(Properties.Resources.PeopleJoeWheatonURL);
        }

        private void lnkWebSite_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
        {
            Process.Start(lnkWebSite.Text);
        }

        private void lnkOnlineHelp_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
        {
            Process.Start(lnkOnlineHelp.Text);
        }

        private void lnkIssues_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
        {
            Process.Start(lnkIssues.Text);
        }
    }
}